src/core/control/zoom/ZoomControl.cpp

```diff
@@ -183,39 +217,18 @@ auto ZoomControl::getVisibleRect() -> Rectangle<double> {
     return layout->getVisibleRect();
 }
 
-void ZoomControl::setScrollPositionAfterZoom(utl::Point<double> scrollPos) {
-    size_t currentPageIdx = this->view->getCurrentPage();
-
-    // To get the layout, we need to call view->getWidget(), which isn't const.
-    // As such, we get the view and determine `unscaledPixels` here, rather than
-    // in `getScrollPositionAfterZoom`.
-    GtkWidget* widget = view->getWidget();
-    Layout* layout = gtk_xournal_get_layout(widget);
-
-    // TODO(personalizedrefrigerator) While this zooms in nearly where intended, it
-    //                                "jitters" while zooming if very far into the document.
-
-    // Not everything changes size as we zoom in/out. The padding, for example,
-    // remains constant!
-    this->unscaledPixels = {static_cast<double>(layout->getPaddingLeftOfPage(currentPageIdx)),
-                            static_cast<double>(layout->getPaddingAbovePage(currentPageIdx))};
-
-    // Use this->zoomWidgetPos to zoom into a location other than the top-left (e.g. where
-    // the user pinched).
-    this->scrollPosition = (scrollPos - this->unscaledPixels + this->zoomWidgetPos) / this->zoom;
-}
-
 auto ZoomControl::getScrollPositionAfterZoom() const -> utl::Point<double> {
     //  If we aren't in a zoomSequence, `unscaledPixels`, `scrollPosition`, and `zoomWidgetPos
     // can't be used to determine the scroll position! Return now.
+    // NOTE: this case should never happen currently.
+    //       getScrollPositionAfterZoom is called from XournalView after setZoom() fired the ZoomListeners
     if (this->zoomSequenceStart == -1) {
```

Comment:
If this case should never happen, make this a mistake (add an `xoj_assert()` or something like that)

Reply:
I'm unsure what is best here:
```c++
    if (this->zoomSequenceStart == -1) {
        xoj_assert_message(true, "ZoomControl::getScrollPositionAfterZoom() was called outside of a zoom sequence. ");
        return {-1, -1};
    }
```
or simply
```c++
    xoj_assert_message(this->zoomSequenceStart == -1, "ZoomControl::getScrollPositionAfterZoom() was called outside of a zoom sequence. ");
```
?